src/bin/sfcp.rs

```diff
@@ -0,0 +1,106 @@
+#[cfg(feature = "sfcp")]
+mod sfcp {
+    use anyhow::{anyhow, Context};
+    use clap::{Parser, Subcommand};
+    use object_store::path::Path;
+    use object_store_ffi::Compression;
+    use std::path::{Path as OsPath, PathBuf};
+    use tracing_subscriber::{layer::SubscriberExt, util::SubscriberInitExt};
+
+    #[derive(Parser, Debug)]
+    #[command(name = "sfcp")]
+    #[command(author, version, about = "Copy tool for Snowflake internal stages", long_about = None)]
+    struct Cli {
+        from: String,
+        to: String,
+        #[arg(short, long)]
+        compression: Option<Compression>
+    }
+
+    #[derive(Debug)]
+    enum Location {
+        Stage(String, Path),
+        File(PathBuf),
+        Directory(PathBuf)
+    }
+
+    fn parse_arg(arg: &str, will_create: bool) -> anyhow::Result<Location> {
+        if arg.starts_with("@") {
+            let (stage, key) = arg
+                .strip_prefix("@")
+                .unwrap()
+                .split_once('/')
+                .context("Missing key of stage location")?;
```

Comment:
Does `context?` return an `Err` too?

Reply:
Yes, it wraps the underlying error this another one that has this context message, but the original error becomes the source of this new one